0006/README.md

```diff
@@ -0,0 +1,318 @@
+    same branch share the same version.
+
+#### Code Contribution
+
+Since the repo uses the Git submodule in a non-standard way and for easy
+building website purposes, people SHOULD NOT use submodules to commit any
+change.
+
+When a community member needs to make a contribution, they fork the repo (and
+clone it locally if they are familiar with git). If the change is for a
+versioned file (a.k.a. spec file), they should make the change in the version
+branch, and send a PR to the version branch. Meanwhile, if the change is for an
+unversioned file or the website itself, they should make the change in `main`,
+and send a PR to `main`.
+
+However, if the change comes from one of the repo owners, and it has to modify
+files in both `main` and version branches in one PR, we assume the owner
+understands the risk and knows how to send a proper code update with Submodule
+changes.
+
+#### Submodule Pointer
+
+A new change in a version branch makes the Submodule pointer in the `main`
+branch stale. A Github Workflow script may be created to automatically update
+the Submodule pointing to the latest commit, when the PR is merged. If desired,
+the repo owner can also manually launch the Github Workflow.
+
+## Automate Building and Deploying The Website
+
+A shell script should assemble the website, instead of using any logic in the
+Github Workflow script.
+
+### Checkout repo, load all submodules, and build once
+
+The GitHub Workflow clones the `main` branch once, pulls the latest files for
+all the submodules, builds the website once, and deploys the website to Github
+Pages.
+
+*Disadvantages*
+
+-   The usage of Git submodule is non-standard, and it leaves some versioned
+    directories in the `main` branch.
+
+*What is changing*
+
+-   To serve the website locally, step 2 in
+    [/docs/README.md](https://github.com/slsa-framework/slsa/blob/main/docs/README.md)
+    is updated to be:
+
+    ```bash
+    git clone --recurse-submodules https://github.com/chtiangg/slsa.git
+    git submodule update --remote
+    cd slsa/docs
+    ```
+-   A new Github Workflow script builds and deploys the website. The following
+    is a cod snapshot.
+    
+    ```yaml
+    # deploy_website.yml lives in .github/workflows.
+    # This script is preliminary and for illustrative purposes.
+
+    # ... Prepares the environment for Github Workflow
+    jobs:
+    build:
+        runs-on: ubuntu-latest
+        steps:
+        - name: Checkout
+            uses: actions/checkout@v3
+            with:
+                submodules: recursive
+        - name: Setup pages
+            uses: actions/configure-pages@v3
+        - name: Build
+            uses: actions/jekyll-build-pages@v1
+            with:
+                source: ./docs
+        - name: Upload artifacts
+            uses: actions/upload-pages-artifact@v1
+    # ... Deploy to your own website.
+    ```
+
+### Alternatives considered: Build each branch separately, then assemble
```

Comment:
Is there a reason why vendoring of the versioned spec wasn't considered?

Reply:
Added an another variant for using a repo to manage a specific version. I will resolve this comment.